benchmarks/scripts/start-agent-and-capture-metrics.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+# Copyright 2014-2020 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Script which starts the agent in the foreground with the provided configuration file, runs it for
+# X minutes, captures relevant agent process level metrics and submits them to CodeSpeed.
+
+# NOTE: This script intentionally runs the agent directly inside the main Circle CI job command
+# container to avoid additional container overhead.
+# Eventually we will want to run those benchmarks / metrics captures on dedicated hosts to ensure
+# consistent behavior and performance across runs.
+# In the mean time, if the performance on Circle CI is too unpredictable, we will need to do multiple
+# run and use a computed percentile value or similar.
+
+# Every command failure (aka non zero exit) in the script should be treated as a fatal error
+set -e
+
+SCRIPT_DIR=$(readlink -f $(dirname ${BASH_SOURCE[0]}))
+
+# Verify mandatory environment variables are set
+if [ -z "${CODESPEED_URL}" ]; then
+    exit 2
```

Comment:
Add error message indicating missing env var.

Reply:
Yep, I also planned to refactor this in a common.sh file so we can re-use this check between both of the scripts.

Reply:
Pushed a change where common functionality is refactored in common.sh file.